src/irreps/dnirrep.jl

```diff
@@ -0,0 +1,279 @@
+"""
+    struct DNIrrep{N} <: AbstractIrrep{Dihedral{N}}
+    DNIrrep{N}(n::Integer, isodd::Bool=false)
+    Irrep[Dihedral{N}](n::Integer, isodd::Bool=false)
+
+Represents irreps of the dihedral group ``D_N = Z_N ⋊ C`` (``Z_N`` and charge conjugation or reflection).
+
+## Properties
+
+- `j::Int`: the value of the ``Z_N`` charge.
+- `isodd::Bool`: the representation of charge conjugation.
+
+Combined these take the values ``+0, -0, 1, ..., (N - 1) / 2`` for odd ``N``, and
+``+0, -0, 1, ..., N / 2 - 1, +(N/2), -(N/2)`` for even ``N``.
+"""
+struct DNIrrep{N} <: AbstractIrrep{Dihedral{N}}
+    # store isodd in right bit, use rest for storing the angle
+    data::UInt8
+    DNIrrep{N}(data::UInt8) where {N} = new{N}(data)
+end
+
+function DNIrrep{N}(j::Integer, isodd::Bool = false) where {N}
+    @assert 0 < N < 32 "DNIrrep requires 0 < N < 32 to function properly"
+    0 <= j <= (N >> 1) ||
+        throw(DomainError(j, "DNIrrep only has irreps 0 <= j <= (N ÷ 2)"))
+    !isodd || j == 0 || (iseven(N) && (j == (N >> 1))) ||
+        throw(DomainError(j, "DNIrrep only has odd irreps when `j == 0` or `iseven(N) && j == N / 2`"))
+    return DNIrrep{N}((j % UInt8) << 1 | isodd)
+end
+
+Base.propertynames(x::DNIrrep) = (:j, :isodd, :data)
+
+function Base.getproperty(a::DNIrrep{N}, sym::Symbol) where {N}
+    if sym === :j
+        return getfield(a, :data) >> 1
+    elseif sym === :isodd
+        return Bool(getfield(a, :data) & true)
+    elseif sym === :data
+        return getfield(a, :data)
+    else
+        error("Unknown property $sym")
+    end
+end
+
+FusionStyle(::Type{DNIrrep{N}}) where {N} = N < 3 ? UniqueFusion() : SimpleFusion()
+sectorscalartype(::Type{DNIrrep{N}}) where {N} = Float64
+Base.isreal(::Type{DNIrrep{N}}) where {N} = true
+
+Base.one(::Type{DNIrrep{N}}) where {N} = DNIrrep{N}(0, false)
+Base.conj(a::DNIrrep) = a
+
+Base.hash(a::DNIrrep, h::UInt) = hash(a.data, h)
+Base.convert(::Type{DNIrrep{N}}, (j, n)::Tuple{Integer, Bool}) where {N} = DNIrrep{N}(j, n)
+
+Base.getindex(::IrrepTable, ::Type{Dihedral{N}}) where {N} = DNIrrep{N}
+
+const D3Irrep = DNIrrep{3}
+const D4Irrep = DNIrrep{4}
+
+function Base.show(io::IO, a::DNIrrep)
+    if get(io, :typeinfo, nothing) !== typeof(a)
+        print(io, type_repr(typeof(a)))
+    end
+    print(io, "(", a.j, ", ", a.isodd, ")")
+    return nothing
+end
+function Base.show(io::IO, ::MIME"text/plain", a::DNIrrep{N}) where {N}
```

Comment:
In which context is this show method used? This would print the sector in a format that is not valid code, right?

Reply:
This would be the one used when printing to strings, because indeed it's not valid code (that's what typically the two-argument version is reserved for).
I think the added value is rather low though, so I simply removed this.